Vagrantfile

```diff
@@ -0,0 +1,100 @@
+job = ENV['CI_JOB'] || ""
+guest_home_dir = '/home/vagrant'
+
+# All Vagrant configuration is done below. The "2" in Vagrant.configure
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure("2") do |config|
+
+  # By default vagrant sync the current directory. Let's disabled it because the directory
+  # will be synced later to the proper destination.
+  config.vm.synced_folder ".", "/vagrant", disabled:true
+  # TODO: Ideally we should be using nfs or 9p for bidirectional syncing
+  config.vm.synced_folder ".", "#{guest_home_dir}/go/src/github.com/kata-containers/tests", type:"rsync"
+  config.vm.synced_folder "#{ENV['GOPATH']}/src/github.com/kata-containers/kata-containers",
+    "#{guest_home_dir}/go/src/github.com/kata-containers/kata-containers", type:"rsync"
+
+  config.vm.provider "libvirt" do |lv|
+    lv.driver = "kvm"
+    lv.cpus = "4"
+    lv.memory = "8192"
+  end
+
+  # Shared provision script.
+  config.vm.provision "shell", env: {"CI_JOB" => job}, inline: <<-SHELL
+
+    export GOPATH="#{guest_home_dir}/go"
+    kata_tests_repo_dir="${GOPATH}/src/github.com/kata-containers/tests"
+
+    env_file="#{guest_home_dir}/ci_job_env"
+    cat <<-EOF > ${env_file}
+export GOPATH="$GOPATH"
+export PATH="/usr/local/go/bin:\$GOPATH/bin:\$PATH"
+export CI="true"
+export CI_JOB="$CI_JOB"
+pushd $kata_tests_repo_dir
+source .ci/ci_job_flags.sh
+popd
+EOF
+
+    source ${env_file}
+
+    # Customize the .bashrc so that it will have the variables exported
+    # after log-in.
+    echo "cd $kata_tests_repo_dir" >> #{guest_home_dir}/.bashrc
+    echo "source $env_file" >> #{guest_home_dir}/.bashrc
+
+    distro=$(source /etc/os-release; echo $ID)
+    case "$distro" in
+      "fedora")
+        # TODO: The kubernetes tests. Maybe it should be handled on
+        # integration/kubernetes/init.sh
+        sudo systemctl stop firewalld || true
+        sudo swapoff -a || true
+
+        # Fedora >= 32 Kernel comes configured with cgroup v2 by default.
+        # This switches back to cgroup v1. It requires a reboot.
+        sudo dnf install -y grubby
+        sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+
+        # It tries to build a ubuntu based image by default. When it pulls the ubuntu image
+        # from docker.io it can hit the pulling ratio error. Let's use fedora.
+        export osbuilder_distro="fedora"
+        ;;
+      "ubuntu")
+        # TODO: redis-server package fails to install if ipv6 is disabled. Move this
+        # code to the setup script.
+        if [[ $(sudo sysctl -n net.ipv6.conf.all.disable_ipv6) -eq 1 ]]; then
+          sudo sysctl -w net.ipv6.conf.all.disable_ipv6=0
+          sudo sed -i 's/\(net.ipv6.conf.all.disable_ipv6\).*/\1=0/' /etc/sysctl.conf
+        fi
+        sudo systemctl enable ssh
```

Comment:
Is this really only required for Ubuntu? Is it actually required? (An Ubuntu server tends to have this enabled I thought?)

Reply:
This is old code that should be removed, sorry. Fedora needs to reboot so the switch to cgroup v1 takes effect; I was rebooting ubuntu as well and encountered problems to reconnect at the VM, I thought it was `sshd` not enabled. This explains why I had that command. Currently only Fedora reboots, so I can get ride of that line.